index.js

```diff
@@ -24,28 +22,33 @@ module.exports = function(mugshot, testRunnerCtx) {
       var msg = composeMessage(message);
 
       Assertion.addProperty(name, function() {
-        var captureItem = this._obj;
-        var deferred = Q.defer();
-        var _this = this;
+        var _this = this,
+            captureItem = this._obj,
+            promise, resolve, reject;
+
+        promise = new Promise(function(res, rej) {
+          resolve = res;
```

Comment:
Welcome back @flore77 😄  While I agree that storing the callbacks here helps reduce the level of indentation, I much prefer nesting the following code inside because it helps make the flow more readable.